src/pages/form-validation/password/index.tsx

```diff
@@ -0,0 +1,472 @@
+        [theme.breakpoints.down('xs')]: {
+            marginTop: theme.spacing(2),
+        },
+    },
+    submitButtonContainer: {
+        width: '100%',
+        paddingTop: theme.spacing(3),
+        paddingBottom: theme.spacing(3),
+        display: 'flex',
+        [theme.breakpoints.down('xs')]: {
+            display: 'none',
+        },
+    },
+    mobileSubmitButtonContainer: {
+        width: '100%',
+        paddingTop: theme.spacing(2),
+        paddingBottom: theme.spacing(2),
+        [theme.breakpoints.up('sm')]: {
+            display: 'none',
+        },
+    },
+}));
+
+type FormError = undefined | null | string;
+type OnChangeHandler = InputProps['onChange'];
+
+export const emailRegex = new RegExp(/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i);
+export const phoneNumberRegex = new RegExp(/^((\(\d{3}\)?)|(\d{3}))([\s-./]?)(\d{3})([\s-./]?)(\d{4})$/);
+export const upperCharRegex = new RegExp(/[A-Z]+/);
+export const lowerCharRegex = new RegExp(/[a-z]+/);
+export const numberRegex = new RegExp(/[0-9]+/);
+export const splCharRegex = new RegExp(/(!|@|#|\$|\^|&)+/);
+
+export const PasswordFormValidation = (): JSX.Element => {
+    const [currentPassword, setCurrentPassword] = useState('');
+    const [currentPasswordError, setCurrentPasswordError] = useState<FormError>();
+    const [newPassword, setNewPassword] = useState('');
+    const [newPasswordError, setNewPasswordError] = useState<FormError>();
+    const [confirmPassword, setConfirmPassword] = useState('');
+    const [confirmPasswordError, setConfirmPasswordError] = useState<FormError>();
+    const [showCurrentPassword, setshowCurrentPassword] = useState<boolean>(false);
+    const [showNewPassword, setShowNewPassword] = useState<boolean>(false);
+    const [showConfirmPassword, setShowConfirmPassword] = useState<boolean>(false);
+    const [passwordErrors, setPasswordErrors] = useState({
+        minLengthRequired: false,
+        atLeast1UpperCharRequired: false,
+        atLeast1LowerCharRequired: false,
+        atLeast1NumberRequired: false,
+        atLeast1SplCharRequired: false,
+    });
+
+    const theme = useTheme();
+    const classes = useStyles(theme);
+    const dispatch = useDispatch();
+    const PASSWORD_MISMATCH = 'Passwords do not match';
+
+    const getPasswordCriteriaIcon = (error: boolean): JSX.Element => (
+        <Done
+            style={{
+                fontSize: 18,
+                color: error ? theme.palette.primary.main : Colors.gray[200],
+                marginRight: theme.spacing(),
+            }}
+        />
+    );
+
+    const validateCurrentPassword = useCallback((): void => {
+        const tempcurrentPassword = currentPassword;
+        let tempcurrentPasswordError = '';
+        if (!tempcurrentPassword.trim()) {
+            tempcurrentPasswordError = 'required';
+        }
+        setCurrentPasswordError(tempcurrentPasswordError);
+    }, [currentPassword, setCurrentPasswordError]);
```

Comment:
All these "setXXX" functions would not be updated and should not appear in this dependency list.

Reply:
All these are removed.